invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/AccessField/components/ProtectionButtons.js

```diff
@@ -8,11 +8,18 @@
 
 import React, { Component } from "react";
 import { Button } from "semantic-ui-react";
-import { Field } from "formik";
+import { FastField } from "formik";
 import { i18next } from "@translations/invenio_rdm_records/i18next";
 import PropTypes from "prop-types";
 
 class ProtectionButtonsComponent extends Component {
+  componentDidMount() {
+    const { formik, disabled, fieldPath } = this.props;
+    if (disabled) {
```

Comment:
maybe add a small comment explaining the case? i.e `disabled` -> user cannot set the record to public? Also, make sure you test that when the user is removing the selected restricted community the access field is updated correctly between `public` and `restricted` and also if they select again a restricted then the value is set to `restricted`. I am just a bit cautious about the FastField and the fact that we set the initial value to `componentDidMount`...

Reply:
Added comment and tested publishing of records for restricted and public communities and both work fine.